Automattic/wp-calypso
# Framework: Reduxify help links

The original help search uses a legacy help links Flux store. This PR aims to get rid of that legacy store and reduxify it in order to be consistent with our recent code best practices.

Since the original Flux store was very small and the new Redux implementation is rather small too, I thought we could do this in a single PR. A list of the changes this PR introduces is:

* Add a simple Redux implementation for help links (actions, data layer, reducer, selector and query component)
* Update the existing `HelpSearch` component to use the Redux implementation.
* Remove the usage of an undefined `search` prop in `HelpSearch`.
* Sort some imports and props in `HelpSearch`.
* Remove unnecessary `displayName` in `HelpSearch`.
* Destructure `translate` for shorter expressions in `HelpSearch`.
* Completely remove the no longer used legacy `lib/help-search` Flux store.

This PR also introduces a small improvement over the previous implementation. Previously, when we search for something and then we erase our search query, an useless network request was issued with an empty search query. This will no longer happen, as our query component will guard for empty search queries and will only issue a request when a search query is present.

You might notice the PR lacks any state tests - this is intentional - any tests would only duplicate the existing code that they test, which we're consciously trying to avoid.

To test:
* Checkout this branch
* Go to http://calypso.localhost:3000/help
* Play with the search form at the top of the page.
* Verify the search form works like it did before, and there are no behavioral or visual changes, and no issues/side effects introduced.
